internal/dataplane/parser/translators/ingress.go

```diff
@@ -142,6 +145,9 @@ func (i *ingressTranslationIndex) Add(ingress *netv1.Ingress, addRegexPrefix add
 					servicePort:      port,
 					addRegexPrefixFn: addRegexPrefix,
 				}
+				// the order of routes ultimately doesn't matter, but using a consistent order simplifies unit testing,
+				// as the output only has array indices and cannot be accessed by name
```

Comment:
That comment might not be 100% accurate because the order ultimately does matter when we want to figure out if the config is different than the one were going to overwrite (in Admin API) right?

Reply:
I believe sorting is handled in another layer - `deckgen.ToDeckContent` sorts all the entities: https://github.com/Kong/kubernetes-ingress-controller/blob/1e541bbe7212b72571fa8411bc17434d4bb5a38c/internal/dataplane/deckgen/generate.go#L38

I think it's okay to stick to that and leave the parser to return non-deterministically ordered slices.